File: basics/구현/왕실의 나이트.cpp
#include <string>

using namespace std;

string now;
int result = 0;
int dx[] = {-2, -2, -1, -1, 1, 1, 2, 2};
int dy[] = {-1, 1, -2, 2, -2, 2, -1, 1};

int main(){
  cin >> now;
  int row = now[1] - '0'; // '1'은 49
  int col = now[0] - 'a' + 1;

  for(int i = 0; i < 8; i++){
    int nrow = row + dx[i];
    int ncol = col + dy[i];

    if(nrow > 0 && nrow < 9 && ncol > 0 && ncol < 9)
      result++;
  }

  cout << result << '\n';
}

File: basics/구현/왕실의 나이트.py
now = input()
row = int(now[1])
col = int(ord(now[0])) - int(ord('a')) + 1
moves = [(-2, -1), (-2, 1), (-1, -2), (-1, 2), (1, -2), (1, 2), (2, -1), (2, 1)]


result = 0
for move in moves:
    nrow, ncol = row + move[0], col + move[1]
    if nrow > 0 and nrow < 9 and ncol > 0 and ncol < 9:
        result += 1

print(result)
